src/views/Search.vue

```diff
@@ -0,0 +1,151 @@
+    div Hostname
+      select(v-model="hostname")
+        option(v-for="hostname in Object.keys($store.getters['buckets/bucketsByHostname'])")
+          | {{hostname}}
+    div Start: {{start.format()}}
+    div End: {{stop.format()}}
+    //div
+      label Use regex
+      input(type="checkbox" v-model="use_regex")
+    div
+      label Filter AFK
+      input(type="checkbox" v-model="filter_afk")
+    //div.form-row
+      div.form-group.col-md-6
+        | Start
+        input.form-control(type="date", :max="today", v-model="startdate")
+      div.form-group.col-md-6
+        | End
+        input.form-control(type="date", :max="tomorrow", v-model="enddate")
+
+    div.form-inline
+
+  div(v-if="status == 'searching'")
+    div #[icon(name="spinner" pulse)] Searching...
+
+  div(v-if="events != null")
+    hr
+
+    div.form-group
+      select.form-control(v-model="vis_method")
+        option(value="eventlist") Event List
+        option(value="timeline") Timeline
+        option(value="summary") Summary
+        option(value="raw") Raw JSON
+
+    div(v-if="vis_method == 'timeline'")
+      aw-timeline(type="simple", :event_type="event_type", :events="events")
+    div(v-if="vis_method == 'eventlist'")
+      aw-eventlist(:events="events")
+    div(v-if="vis_method == 'summary'")
+      input.form-control(type="text" v-model.lazy.trim="summaryKey" placeholder="data key" style="margin-bottom: 1em;")
+      aw-summary(:fields="events", :colorfunc="colorfunc", :namefunc="namefunc")
+    div(v-if="vis_method == 'raw'")
+      pre {{ events }}
+
+    div
+      | Didn't find what you were looking for?
+      br
+      | Add a week to the search: #[b-button(size="sm" variant="outline-dark" @click="start = start.subtract(1, 'week'); search()") +1 week]
+
+</template>
+
+<style scoped lang="scss"></style>
+
+<script>
+import _ from 'lodash';
+import moment from 'moment';
+import { canonicalEvents } from '~/queries';
+
+import 'vue-awesome/icons/search';
+import 'vue-awesome/icons/spinner';
+import 'vue-awesome/icons/angle-double-down';
+import 'vue-awesome/icons/angle-double-up';
+
+export default {
+  name: 'Search',
+  data() {
+    return {
+      pattern: '',
+      vis_method: 'eventlist',
+      event_type: 'currentwindow',
+      events: null,
+      error: '',
+      status: null,
+      hostname: '',
+
+      // Options
+      show_options: false,
+      use_regex: true,
+      filter_afk: true,
+      start: moment().subtract(1, 'day'),
+      stop: moment().add(1, 'day'),
+
+      /* Summary props */
+      summaryKey: 'title',
+      colorfunc: null,
+      namefunc: null,
+    };
+  },
+  mounted: async function () {
+    this.colorfunc = this.summaryKeyFunc;
+    this.namefunc = this.summaryKeyFunc;
+    await this.$store.dispatch('buckets/ensureBuckets');
+    this.hostname = Object.keys(this.$store.getters['buckets/bucketsByHostname'])[0];
+  },
+  methods: {
+    search: async function () {
+      let query = canonicalEvents({
```

Comment:
Using canonicalEvents limits us to only delete events in the window bucket, I think it's better to drop the filter_afk support and just write a new query that simply fetches all events from a bucket and possibly categorizes them.

Reply:
See https://github.com/ActivityWatch/aw-webui/pull/236#discussion_r537004344